src/job/file_list_dump.rs

```diff
@@ -0,0 +1,459 @@
+            Arc::new(field_flattened.finish()),
+            Arc::new(field_min_ts.finish()),
+            Arc::new(field_max_ts.finish()),
+            Arc::new(field_records.finish()),
+            Arc::new(field_original_size.finish()),
+            Arc::new(field_compressed_size.finish()),
+            Arc::new(field_index_size.finish()),
+        ],
+    )?;
+    Ok(batch)
+}
+
+async fn generate_dump(
+    org: &str,
+    stream: &str,
+    range: (i64, i64),
+    files: Vec<FileRecord>,
+    job_id: i64,
+) -> Result<(), anyhow::Error> {
+    // if there are no files to dump, no point in making a dump file
+    if files.is_empty() {
+        return Ok(());
+    }
+
+    let config = get_config();
+    let batch_size = files.len();
+
+    let mut fkeys = Vec::with_capacity(files.len());
+    // we only delete if dual write is disabled, so we only create the entries if it is disabled
+    if !config.common.file_list_dump_dual_write {
+        files.iter().for_each(|f| {
+            fkeys.push(FileKey {
+                key: "files/".to_string() + &f.stream + "/" + &f.date + "/" + &f.file,
+                meta: f.into(),
+                deleted: true,
+                segment_ids: None,
+            })
+        });
+    }
+
+    let mut buf = Vec::new();
+    let mut writer = get_writer(FILE_LIST_SCHEMA.clone(), &mut buf)?;
+    // we split the file list into vec of vecs, with each sub-vec having PARQUET_BATCH_SIZE elements
+    // at most (last may be shorter) doing this the following way means we don't have to clone
+    // anything, we simply shift the ownership via iterator
+    let chunks: Vec<Vec<_>> = files
+        .into_iter()
+        .chunks(PARQUET_BATCH_SIZE)
+        .into_iter()
+        .map(|c| c.collect())
+        .collect();
+    for chunk in chunks {
+        let batch = create_record_batch(chunk)?;
+        writer.write(&batch).await?;
+    }
+    writer.close().await?;
+
+    let ts = chrono::DateTime::from_timestamp_micros(range.0).unwrap();
+
+    let file_name = format!(
+        "{:04}/{:02}/{:02}/{:02}/{}.parquet",
+        ts.year(),
+        ts.month(),
+        ts.day(),
+        ts.hour(),
+        config::ider::generate()
+    );
+
+    let stream_key = get_dump_stream_key(org, stream);
+    let file_key = format!("files/{stream_key}/{file_name}");
+
+    let meta = FileMeta {
+        min_ts: range.0,
+        max_ts: range.1,
+        records: batch_size as i64,
+        original_size: buf.len() as i64,
+        compressed_size: buf.len() as i64,
+        index_size: 0,
+        flattened: false,
+    };
+
+    fkeys.push(FileKey {
+        key: file_key.clone(),
+        meta: meta.clone(),
+        deleted: false,
+        segment_ids: None,
+    });
+
+    // first store the file in storage
+    // then update the entries in db,
+    // and if both pass only then set the job as dumped=true
+
+    infra::storage::put(&file_key, buf.into()).await?;
+    infra::file_list::batch_process(&fkeys).await?;
```

Comment:
we can't use `batch_process` here, it will deleted all the items from object_store which marked `deleted`.

Reply:
Sorry, here no problem, but like you said, after this, we don't need calculate stream stats from file_list_dump